src/widgets/modals/CashTransactionModal.js

```diff
@@ -120,8 +114,41 @@ export const CashTransactionModal = ({ onConfirm }) => {
     setIsDescriptionModalOpen(false);
   };
 
+  const toggleTransactionType = () => setIsCashIn(!isCashIn);
+
+  const toggles = useMemo(
+    () => [
+      { text: 'Cash in', onPress: toggleTransactionType, isOn: isCashIn },
+      { text: 'Cash out', onPress: toggleTransactionType, isOn: !isCashIn },
+    ],
+    [isCashIn]
+  );
+
+  const PressReason = useCallback(
+    () =>
+      isCashIn ? (
+        <TouchableOpacity style={localStyles.containerStyle}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{placeholderReasonDisabled}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle} />
+        </TouchableOpacity>
+      ) : (
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressReason}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{reason?.title ?? placeholderReason}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <ChevronDownIcon />
+          </View>
+        </TouchableOpacity>
+      ),
+    [isCashIn, reason]
```

Comment:
feel like this could just as nicely be `{isCashIn && <Touch.... >}` to just not show the reason selection for cash outs